### DISCUSSION TITLE
Add flag to only sync LDAP role on first login

### DISCUSSION THREAD

**Discussion by LasseRosenow** [directus/directus#12765] [Feature Requests]
### Describe the Bug

I have LDAP connected to my directus, but whenever I login using LDAP, my the role of my user gets reset.

In our system we do want to have the ability to promote certain user only within directus to give them more rights.
But we don't do that within LDAP as the cms team is not allowed to change `active directory groups` of users.

So we just change the role of LDAP users that need to have more rights.

But whenever those user logout and login again, their role gets reset.

### To Reproduce

1. Connect directus to LDAP
2. Login with LDAP user
3. Change its role
4. Logout with LDAP user
5. Login again
6. The role should have reset now

### What version of Directus are you using?

9.8.0

### What version of Node.js are you using?

16

### What database are you using?

MariaDB

### What browser are you using?

Chrome, Firefox

### How are you deploying Directus?

On a remote Debian server

**Comment by rijkvanzanten** [comment]
Thoughts @aidenfoxx?

**Comment by aidenfoxx** [comment]
@rijkvanzanten This is working as intended. We sync user roles whenever you log in with LDAP, so if you don't belong to a group in AD that matches a role in Directus, you get no role.

@LasseRosenow If you want to promote a user, you should do so in AD. From my perspective the point of LDAP is that you can manage your users from AD and not have to maintain multiple platforms.

**Comment by aidenfoxx** [comment]
A configuration flag could be added so that we only set user roles when creating an LDAP user for the first time, and then defer role management to Directus thereafter. But I believe it should be disabled by default.

**Comment by LasseRosenow** [comment]
@aidenfoxx 

Okay I can see your point, but for our use case as we are not allowed to change ldap rules, we need to be able to manually set rules in ldap.

Our LDAP use-case basically is just to have a consistent login, but the permisson/role management should just be done inside directus.

So having a flag to disable the automatic sync of ldap roles with directus would be perfect :)

**Comment by LasseRosenow** [comment]
Off topic: So as far as I understand it, the active directory group => directus role integration is supposed to work so that the active directory group is the single point of truth?

This makes much sense of course, but wouldn't it be "dangerous" to only sync roles on login? Because some people might stay logged in forever and would thus keep their old role even though their group was removed in the active directory?

**Reply by aidenfoxx** [reply]
The AD group should certainly be the point of truth. As you say, it is currently possible for the role to last forever if the user is never logged out since we only sync on login. We do have some simpler checks after login to continuously verify that the account is still active and valid, but we don't check the role.

Wondering what the best behavior would be in the scenario that a role changes while the user is logged in. Probably best to log them out and force a re-auth so we don't break accountability or something @rijkvanzanten?

**Reply by rijkvanzanten** [reply]
> Wondering what the best behavior would be in the scenario that a role changes while the user is logged in. Probably best to log them out and force a re-auth so we don't break accountability or something @rijkvanzanten?

The role ID is pulled from the access token payload, so the next session refresh cycle the new role will kick in automatically (one of the reasons the TTL of the access tokens is so short). If we can sync the roles from LDAP on refresh as well as login (if we're not already), we should be in an OK spot 👍🏻

**Comment by rijkvanzanten** [comment]
Thanks for confirming @aidenfoxx! I thought it was synced, but didn't want to conclude without checking 🙂 

I'll move this to a feature request discussion 👍🏻

**Comment by aidenfoxx** [comment]
@LasseRosenow This should solve your issues https://github.com/directus/directus/pull/12767. Once released you just need to omit a `_GROUP_DN` configuration and roles will no longer sync with AD. You can also specify a `_DEFAULT_ROLE_ID` which will be applied when LDAP users are first created.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
